docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,237 @@
+
+
+.. _interoperable-causality:
+
+Causality for Interoperable Ledgers
+***********************************
+
+The :ref:`ordering guarantees <order-consistency>` between the local ledger and the transaction service does not need to be changed for interoperability.
+The order consistency of several local ledgers and the shared ledger, however, need to be generalized for interoperability.
+
+The virtual shared ledger
+=========================
+
+While every DAML Ledger may keep a physical copy of its shared ledger,
+there is no place that records the result of the interoperability protocol between those ledgers;
+it merely ensures that the individual shared ledgers are consistent.
+Like for local ledgers, consistency is formalized by the existence of a *virtual* shared ledger.
+
+Definition »virtual shared ledger«
+  A **virtual shared ledger** for a set `X` of interoperable DAML Ledgers
+  is a finite labelled directed acyclic graph of transactions that satisfies the following conditions:
+
+  * It satisfies the conditions for a shared ledger.
+
+  * It does not contain **Enter** nor **Leave** actions.
+
+  * Every action in the vertex transactions is labelled with a DAML Ledger from `X`.
+
+Since a cross-ledger transaction can use contracts from different DAML Ledgers,
+the :ref:`projection <da-model-projections>` of transactions needs to be aware of the ledgers.
+In the `PaintOffer` workflow, e.g., Alice's and the painter's projections of the acceptance transactions are the whole transaction
+as they are both stakeholders on the `PaintOffer` contract.
+When this transaction is run across two ledgers, say one ledger for dealing with `Iou`\ s and one ledger for dealing with painting,
+Alice's and the painter's participants on the `Iou` ledger will see only the `Iou` part of the transaction.
+Accordingly, the ledger-aware projections look as follows, where yellow represents the `Iou` ledger and blue stands for the painting ledger.
+Here, the projections to the blue ledger include the actions of the yellow ledger because a projection includes the subactions.
+
+.. https://www.lucidchart.com/documents/edit/f8ec5741-7a37-4cf5-92a9-bf7b3132ba8e
+.. image:: ./images/projecting-transactions-paint-offer-ledger-aware.svg
+   :align: center
+   :width: 60%
+
+Definition »Y-labelled action«
+  An action is **Y-labelled** for a set `Y` if it is labelled with an element of `Y`.
+           
+Definition »ledger-aware projection«
+  Let `tx` be a transaction according to the DAML Ledger Model whose actions are `X`\ -labelled for a set `X` of DAML Ledgers
+  and let `Y` be a subset of `X`.
+  Let `Act` be the set of `Y`\ -labelled subactions of `tx` that have `A` as an informee.
+  The **Y-projection** of `tx` to a party `A` consists of maximal elements of `Act` (w.r.t. the subaction relation) in execution order.
+
+Ledger-aware projections extend to the virtual shared ledger straightforwardly.
+  
+Interoperable order consistency
```

Comment:
As with local ledgers, I wonder whether we shouldn't use the totally ordered Shared Virtual Ledger as a kind of universal object here into which everything embeds.